src/main/java/com/example/gradecalculator/entities/GradeType.java

```diff
@@ -3,38 +3,59 @@
 import jakarta.persistence.*;
 
 @Entity
-@Table(name = "grade_type")
+@Table( name = "grade_type")
 public class GradeType {
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    @Column(name = "id")
     private Long id;
-
-    @Column(name = "name")
+    @Column( unique = true)
     private String name;
 
-    public GradeType() {
-    }
+    private String description;
+    private int weightage;
```

Comment:
Is it best to use an INT in this case? Weightage of a grade can only be a multiple of itself like x * 1, x * 2, but edge-cases like the grade only counts as half a "normal" grade would not be feasbil. 
Consider using a data type that offers more variety

Reply:
You are totally right, I would change it to percentage.